Algorithm improvements

Aeunal opened the pull request:
Implemented the second part of the algorithm and made a few changes on code structure.
'backend/package-lock.json' changed by itself.

ameerkabir commented on backend/src/helper/index.js:
```diff
@@ -11,6 +11,21 @@ export function findInObject(data, searchParameter, exclude){
+    })
+}
+
+export function mergeObject(data, searchParameter1, searchParameter2) {
```
well done, you can use Object.assign() to merge two objects or {...object1, ...Object2} which are native to javascript so you will not need to create your own method to handle that

ameerkabir commented on backend/src/helper/index.js:
```diff
@@ -11,6 +11,21 @@ export function findInObject(data, searchParameter, exclude){
+    return data;
+}
+
 export function fetchData(req) {
```
I think we can all the data from the req.body so you can just do const {country, name , age , basically, anything from the request} = req.body, this way you will not need to reassign them again in a function. and I believe you can rename them if you want as well, like `const { name: fullName}`

ameerkabir replied:
@Aeunal also, there is a conflict if you happy to fix it is fine otherwise I can go ahead and fix the conflict

ameerkabir commented on backend/src/controllers/index.js:
```diff
@@ -1,28 +1,26 @@
 
 export const homepage = (req, res) => {
-  console.log('what', req.body);
+  console.log('Homepage:', req);
```
```suggestion
```

Aeunal commented:
- Changed all the algorithm parts with their Ramda counterparts.
- Removed unused methods
- Moved the main algorithm from /controllers to /helper directory *(to decrease the number of imports)*
- **Implemented the third part of the algorithm**

ameerkabir commented on backend/src/helper/index.js:
```diff
@@ -1,47 +1,151 @@
+  return [];
+}
+
+export async function filterResponse(opportunities, relation, key, value, exclude = false) {
```
@Aeunal  I wonder if we could just use `propEq` instead of sending it as a parameter because it's a globally imported hence you might need to reinvent the wheel I 🤔, but let me know  I am curious to know.

Aeunal replied:
Maybe we can but I think we would have to make it as a new method like `filterEqualElements`. Because sometimes I use `filterResponse` to get the elements from the object array that has the same value with the value I gave as a parameter, then I am sending `propEq` as the relation. But if I want to get the elements that have a lower value than I sent, then I am sending `propSatisfies` to check values. I made it this way so the usage of the method would be extendable.

ameerkabir replied:
@Aeunal, in that case, it makes sense to me to use relation as you did.
